src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -369,6 +370,13 @@ private bool StartCore(ProcessStartInfo startInfo)
                 (userId, groupId) = GetUserAndGroupIds(startInfo);
             }
 
+            // .NET applications don't echo characters unless there is a Console.Read operation.
+            // Unix applications expect the terminal to be in an echoing state by default.
+            // To support processes that interact with the terminal (e.g. 'vi'), we need to configure the
+            // terminal to echo. We keep this configuration as long as there are interactive applications.
+            // Based on ProcessStartInfo we determine if the child may be an interactive application.
```

Comment:
It would be nice if we had something like `ProcessStartInfo.IsInteractiveApplication`. Now we are looking at `RedirectStandardInput/Output`. This will put the terminal into an echoing state more than we should because the user had no reason to be changing those for non-interactive applications (until now).

Reply:
With https://github.com/dotnet/corefx/issues/35685 we could improve this check.